tests/functional/u/used/used_before_assignment_py311.py

```diff
@@ -1,6 +1,12 @@
 """assert_never() introduced in 3.11"""
 from enum import Enum
-from typing import assert_never
+
+from pylint.constants import PY311_PLUS
+
+if PY311_PLUS:
```

Comment:
I think we should use sys.version_info directly here. Importing from pylint create coupling so more maintenance work when we drop python 3.11 and more importantely code in the wild will never import out internal constants so it makes functional tests less likely to be representative of real code.

Reply:
Why is it Less maintenance?

Reply:
We'll have to modify this when we drop python 3.11 and the constant do not exists anymore, right ?

Reply:
Yes, but I was suggesting it was the same whether we use the constant or the version guard.

Reply:
This is a functional test, I don't think functional test should look like internal pylint code. However it's not really important, let's move on